### PR TITLE
Support Fedora 24 (redux)

### PR DESCRIPTION
**Pull request by andyleejordan**
This reapplies #2738 with a bug fix for #2970.

@paulcallen you'll need to rebase your work-in-progress again if this gets in first.

I tested package creation and installation on the Ubuntu 16.04 Docker image:
```
PS /> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      6.0.0-alpha
PSEdition                      Core
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   3.0.0.0
GitCommitId                    v6.0.0-alpha.14-27-g13f01a651a4561690921b2a8d5fdc4762b339589
CLRVersion
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

and the CentOS 7 image:
```
PS /PowerShell> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      6.0.0-alpha
PSEdition                      Core
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   3.0.0.0
GitCommitId                    v6.0.0-alpha.14-27-g13f01a651a4561690921b2a8d5fdc4762b339589
CLRVersion
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

### CODE REVIEW

File: build.psm1 [hunk 1]
```diff
@@ -31,6 +31,14 @@ if ($IsLinux) {
     $IsUbuntu14 = $IsUbuntu -and $LinuxInfo.VERSION_ID -match '14.04'
     $IsUbuntu16 = $IsUbuntu -and $LinuxInfo.VERSION_ID -match '16.04'
     $IsCentOS = $LinuxInfo.ID -match 'centos' -and $LinuxInfo.VERSION_ID -match '7'
+    $IsFedora = $LinuxInfo.ID -match 'fedora' -and $LinuxInfo.VERSION_ID -ge 24
+    $IsRedHatFamily = $IsCentOS -or $IsFedora
+
+    # Workaround for temporary LD_LIBRARY_PATH hack for Fedora 24
```
Comment (vors) [inline build.psm1]: Please add a reference to the discussion on the GitHub for the context

File: docker/release/fedora24/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,64 @@
+FROM fedora:24
+MAINTAINER Andrew Schwartzmeyer <andschwa@microsoft.com>
+
+# This release Dockerfile uses the CentOS 7 build for the purposes of enabling
```
Comment (vors) [inline docker/release/fedora24/Dockerfile]: That should probably go away after the next release, correct?

**Review by vors — APPROVED** [review]

File: build.psm1 (hunk 1 above)
Comment (andschwa) [inline build.psm1]: This is owned by @randomvariable: can you open a PR on top of this to fix up the comments? (Or do it with the documentation additions.)

File: docker/release/fedora24/Dockerfile (hunk 2 above)
Comment (andschwa) [inline docker/release/fedora24/Dockerfile]: Hm I'm not sure why my name is on this Dockerfile as maintainer, it should be @randomvariable's probably.

File: build.psm1 [hunk 3]
```diff
@@ -31,6 +31,14 @@ if ($IsLinux) {
...
+    if (Test-Path ENV:\LD_LIBRARY_PATH) {
```
Comment (mirichmo) [inline build.psm1]: The comment implies that this applies only to Fedora 24. Should we add an additional condition to the IF statement?
`
if ($IsFedora -AND Test-Path ENV:LD_LIBRARY_PATH) {
`

**Comment by mirichmo** [comment]
And thanks for addressing the issue @andschwa

File: build.psm1 (hunk 3 above)
Comment (andschwa) [inline build.psm1]: /cc @randomvariable

**Comment by andschwa** [comment]
I'm not seeing any activity from @randomvariable, so I updated his changes per the comments.

**Comment by andschwa** [comment]
I've got an in-person merge approval from @lzybkr, I'll get this finished.

**Pull request merged**